ipa-core/src/protocol/ipa_prf/shuffle/malicious.rs

```diff
@@ -27,16 +29,110 @@ use crate::{
     seq_join::seq_join,
 };
 
+/// This function executes the maliciously secure shuffle protocol on the input: `shares`.
+///
+/// ## Errors
+/// Propagates network, multiplication and conversion errors from sub functions.
+///
+/// ## Panics
+/// Panics when `S::Bits + 32 != B::Bits` or type conversions fail.
```

Comment:
It would be clearer to check this property on entry to `malicious_shuffle`, rather than relying on the eventual call to `concatenate_row_and_tags` to check this property -- this captures the API contract in one place even if the implementation gets refactored in the future.